docs/edgeql/statements/delete.rst

```diff
@@ -21,7 +30,53 @@ DELETE
     statement.  See :ref:`ref_eql_with` for more information.
 
 :eql:synopsis:`DELETE <expr>`
-    Remove objects returned by *expr* from the database.
+    Form the base of the set to be deleted. This base set can be
+    further trimmed with other clauses.
+
+:eql:synopsis:`FILTER <filter-expr>`
+    The optional ``FILTER`` clause, where :eql:synopsis:`<filter-expr>`
+    is any expression that has a result of type :eql:type:`bool`. The
+    condition is evaluated for every element in the base set produced
+    by the ``DELETE`` clause.  The result of the evaluation of the
```

Comment:
> produced
     by the ``DELETE`` clause

This is ambiguous. 

I guess you've just copy/pasted docs from the SELECT statement. We don't want that kind of duplication. I suggest to do it another way: copy/paste the synopsis and simply refer to the SELECT statement for the explanation of how different clauses work.  First paragraph should mention that `DELETE ...` is a sugar for `DELETE (SELECT ...)`; that should be enough.

Reply:
I've reworded that section.